src/irc/core/lag.c

```diff
@@ -48,18 +48,18 @@ static void lag_ping_error(IRC_SERVER_REC *server)
 static void lag_event_pong(IRC_SERVER_REC *server, const char *data,
 			   const char *nick, const char *addr)
 {
-	GTimeVal now;
+	gint64 now;
 
 	g_return_if_fail(data != NULL);
 
-	if (server->lag_sent.tv_sec == 0) {
+	if (server->lag_sent == 0) {
 		/* not expecting lag reply.. */
 		return;
 	}
 
-	g_get_current_time(&now);
-	server->lag = (int) get_timeval_diff(&now, &server->lag_sent);
-	memset(&server->lag_sent, 0, sizeof(server->lag_sent));
+	now = g_get_real_time();
+	server->lag = now - server->lag_sent;
```

Comment:
```suggestion
	server->lag = (now - server->lag_sent) / G_TIME_SPAN_MILLISECOND;
```

Found the issue! It wasn't some uninitialized nonsense, it was storing microseconds in field that expects milliseconds!

Reply:
thanks!